vortex-file/src/read/stream.rs

```diff
@@ -84,6 +88,7 @@ impl<R: VortexReadAt + Unpin> VortexFileArrayStream<R> {
     }
 
     pub fn dtype(&self) -> &DType {
+        // FIXME(ngates): why is this allowed to unwrap?
```

Comment:
This is allowed to unwrap because the caller will know whether it will throw. I have a change that wil get rid of the error here